datadog/resource_datadog_dashboard.go

```diff
@@ -4531,6 +4531,258 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
 	return terraformDefinition
 }
 
+func getFormulaSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"formula_expression": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "String expression built from queries, formulas and functions.",
+				},
+				"limit": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Options for limiting results returned.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"count": {
+								Type:        schema.TypeInt,
+								Optional:    true,
+								Description: "Number of results to return",
+							},
+							"order": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "desc",
+								Description: "Direction of sort.",
+							},
+						},
+					},
+				},
+				"alias": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Expression alias.",
+				},
+			},
+		},
+	}
+}
+
+func getFormulaQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"metric_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions metrics query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "metrics",
+								Description: "Data source for metrics queries.",
+							},
+							"query": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Metrics query definition.",
+							},
+							"aggregator": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "The aggregation methods available for metrics queries.",
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "Name of the query for use in formulas.",
+							},
+						},
+					},
+				},
+				"event_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions events query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Data source for event platform-based queries.",
```

Comment:
Can we do the same `Optional: true, Default: <>` for this data source and the process_query one?

Reply:
Event-based queries can take on a [multitude of values](https://github.com/DataDog/dogweb/blob/4c1c9c70ee08bdbe4ea8df7eafc25eddbbd23a93/dogweb/lib/validation/schemas/query.py#L20).

Process queries currently only support the `process` data source, but will also support `container` as well: https://github.com/DataDog/dogweb/blob/4c1c9c70ee08bdbe4ea8df7eafc25eddbbd23a93/dogweb/lib/validation/schemas/query.py#L81.

For the time being, I think we should leave `process` queries as is (with no change to event-based queries).